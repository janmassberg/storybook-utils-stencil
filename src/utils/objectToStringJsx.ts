import { objectToString } from "./objectToString";

/**
 * Converts any object to a jsx string.
 * @param value {string|number|boolean|function|{}}
 */
export const objectToStringJsx = (value: any): string => {
    if (typeof value === "string") {
        return `"${value}"`;
    }
    if (typeof value === "function") {
        return `{${value.toString()}}`;
    }
    return `{${objectToString(value)}}`;
};
