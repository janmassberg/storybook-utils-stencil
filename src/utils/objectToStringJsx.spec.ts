import { objectToStringJsx } from "./objectToStringJsx";

describe("objectToStringJsx", () => {
    it("returns strings with quotes", () => {
        expect(objectToStringJsx("test-string")).toEqual("\"test-string\"");
    });

    it("should convert numbers correctly", () => {
        expect(objectToStringJsx(1234)).toEqual("{1234}");
    });

    it("should convert boolean values correctly", () => {
        expect(objectToStringJsx(true)).toEqual("{true}");
        expect(objectToStringJsx(false)).toEqual("{false}");
    });

    it("should convert functions correctly", () => {
        expect(objectToStringJsx((event: Event) => {
            console.log(event);
        })).toMatchSnapshot();
    });

    it("should correctly convert objects to jsx", () => {
        expect(objectToStringJsx({
            testString: "test string",
            testNumber: 1234,
            testTrue: true,
            testFalse: false,
        })).toMatchSnapshot();
    });
});
