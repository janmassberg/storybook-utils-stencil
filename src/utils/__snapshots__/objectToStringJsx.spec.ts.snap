// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objectToStringJsx should convert functions correctly 1`] = `
"{function (event) {
            console.log(event);
        }}"
`;

exports[`objectToStringJsx should correctly convert objects to jsx 1`] = `"{{testString: \\"test string\\", testNumber: 1234, testTrue: true, testFalse: false}}"`;
